### PR TITLE
Event Repeat Renaming + Relogic, Event Repeat Test, Default None for Empty Event Types in All RSVP

### PR DESCRIPTION
**Pull request by ochan1**
Event Repeat Renaming:
image

In this PR:
* Event Repeat Renaming + Relogic
* Event Repeat Test
* Default None for Empty Event Types in All RSVP (mainly for the Tabular Event test)

(Does not fix test failures from incorrect Response Code)

### CODE REVIEW

File: hknweb/events/tests/views/event_transactions/test_repeat_event.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+from hknweb.events.tests.views.event_transactions.utils import setUp
+from hknweb.events.utils import generate_recurrence_times
+
+class RepeatIntervalsUtilsTests(TestCase):
+    def test_repeat_time_intervals(self):
+        start_time = datetime(2022, 3, 15, 11, 30)
+        end_time = datetime(2022, 3, 15, 3, 30)
+
+        def base_tests():
```
Comment (bri25yu) [inline hknweb/events/tests/views/event_transactions/test_repeat_event.py:14]: What's the reasoning behind wrapping this in base_tests?

**Comment by bri25yu** [comment]
Also do we want to keep the event recurrence code? Not sure if anyone actually uses it (looking at our calendar).

File: hknweb/events/tests/views/event_transactions/test_repeat_event.py (hunk 1 above)
Comment (ochan1) [reply hknweb/events/tests/views/event_transactions/test_repeat_event.py:14]: Nothing special, just grouping them as a name of a series of small base case sanity checks

Comment (bri25yu) [reply hknweb/events/tests/views/event_transactions/test_repeat_event.py:14]: Why not just make them a test case or unwrap them?

Comment (ochan1) [reply hknweb/events/tests/views/event_transactions/test_repeat_event.py]: Agreed on making it a separate one, just did it

**Comment by ochan1** [comment]
I agree it will be available rarely, but I think some uses cases include the LEG DAY event the towards the end of the Spring 2022 semester + Brownbag Talks in the Fall 2021 semester

Case example too is that I don't use recurring events often on Google Calendar, but it is super helpful when I make weekly events or something for meetings (especially CompServ meetings during my Fall 2021 and Spring 2022 semester)

My opinion is to keep it there and people can use it or skip it, though I do agree it's usage is rare

**Comment by bri25yu** [comment]
Also, if we're gonna be putting the committee meetings on the calendar, we actually have a use case for the feature, I'm down to merge this in

**Pull request merged**